docs/src/man/fusiontrees.md

```diff
@@ -0,0 +1,435 @@
+the ``Z`` isomorphism in the fusion part. Furthermore, the presence of the ``Z``
+isomorphisms does not affect the orthonormality.
+
+We represent splitting trees and their adjoints using a specific immutable type called
+`FusionTree` (which actually represents a splitting tree, but fusion tree is a more common
+term), defined as
+```julia
+struct FusionTree{I<:Sector,N,M,L,T}
+    uncoupled::NTuple{N,I}
+    coupled::I
+    isdual::NTuple{N,Bool}
+    innerlines::NTuple{M,I} # fixed to M = N-2
+    vertices::NTuple{L,T} # fixed to L = N-1
+end
+```
+Here, the fields are probably self-explanotary. The `isdual` field indicates whether an
+isomorphism is present (if the corresponding value is `true`) or not. Note that the field
+`uncoupled` contains the sectors coming out of the splitting trees, before the possible
+``Z`` isomorphism, i.e. the splitting tree in the above example would have
+`sectors = (a₁, a₂, a₃, a₄)`. The `FusionTree` type has a number of basic properties and
+capabilities, such as checking for equality with `==` and support for
+`hash(f::FusionTree, h::UInt)`, as splitting and fusion trees are used as keys in look-up
+tables (i.e. `AbstractDictionary` instances) to look up certain parts of the data of a
+tensor. The type of `L` of the vertex labels can be `Nothing` when they are not needed
+(i.e. if `FusionStyle(I) isa MultiplicityFreeFusion`).
+
+`FusionTree` instances are not checked for consistency (i.e. valid fusion rules etc) upon
+creation, hence, they are assumed to be created correctly. The most natural way to create
+them is by using the `fusiontrees(uncoupled::NTuple{N,I}, coupled::I = one(I))` method,
+which returns an iterator over all possible fusion trees from a set of `N` uncoupled
+sectors to a given coupled sector, which by default is assumed to be the trivial sector of
+that group or fusion category (i.e. the identity object in categorical nomenclature). The
+return type of `fusiontrees` is a custom type `FusionTreeIterator` which conforms to the
+complete interface of an iterator, and has a custom `length` function that computes the
+number of possible fusion trees without iterating over all of them explicitly. This is best
+illustrated with some examples
+
+```@repl sectors
+s = Irrep[SU₂](1/2)
+collect(fusiontrees((s,s,s,s)))
+collect(fusiontrees((s,s,s,s,s), s, (true, false, false, true, false)))
+iter = fusiontrees(ntuple(n->s, 16))
+sum(n->1, iter)
+length(iter)
+@elapsed sum(n->1, iter)
+@elapsed length(iter)
+s2 = s ⊠ s
+collect(fusiontrees((s2,s2,s2,s2)))
+```
+Note that `FusionTree` instances are shown (printed) in a way that is valid code to
+reproduce them, a property which also holds for both instances of `Sector` and instances of
+`VectorSpace`. All of those should be displayed in a way that can be copy pasted as valid
+code. Furthermore, we use contact to determine how to print e.g. a sector. In isolation,
```

Comment:
I might be missing something, but what does "we use contact" mean?

Reply:
Not sure how that word got in there, and what I had in mind instead. This is old docs that I did not (yet) change within this PR.

Reply:
Got it: this is supposed to be "context". 

We use context to determine how to print ...

(corresponding to the `context` keyword in Julia's print methods).